pandera/backends/pandas/error_formatters.py

```diff
@@ -56,10 +65,40 @@ def format_vectorized_error_message(
         "pyspark.pandas"
     ):
         failure_cases = reshaped_failure_cases.failure_case.to_numpy()
-        failure_cases_string = ", ".join(failure_cases.astype(str))
+        total_failures = len(failure_cases)
+
+        if max_failure_cases != -1:
+            if max_failure_cases == 0:
+                failure_cases_string = f"... {total_failures} failure cases"
+            elif total_failures > max_failure_cases:
+                failure_cases_limited = failure_cases[:max_failure_cases]
+                failure_cases_string = ", ".join(
+                    failure_cases_limited.astype(str)
+                )
+                omitted_count = total_failures - max_failure_cases
+                failure_cases_string += f" ... and {omitted_count} more failure cases ({total_failures} total)"
+            else:
+                failure_cases_string = ", ".join(failure_cases.astype(str))
+        else:
+            failure_cases_string = ", ".join(failure_cases.astype(str))
     else:
         failure_cases = reshaped_failure_cases.failure_case
-        failure_cases_string = ", ".join(failure_cases.apply(str))
+        total_failures = len(failure_cases)
+
+        if max_failure_cases != -1:
+            if max_failure_cases == 0:
+                failure_cases_string = f"... {total_failures} failure cases"
+            elif total_failures > max_failure_cases:
+                failure_cases_limited = failure_cases.iloc[:max_failure_cases]
+                failure_cases_string = ", ".join(
+                    failure_cases_limited.apply(str)
+                )
+                omitted_count = total_failures - max_failure_cases
+                failure_cases_string += f" ... and {omitted_count} more failure cases ({total_failures} total)"
+            else:
+                failure_cases_string = ", ".join(failure_cases.apply(str))
+        else:
+            failure_cases_string = ", ".join(failure_cases.apply(str))
```

Comment:
This logic looks like it contains duplication from above. Either consolidate into a single set of logic, or extract into its own function, and call it here. (Think of what's easiest to test)